lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -166,7 +166,7 @@ def parse(body)
       end
 
       def success_from(response)
-        response["responseCode"] == "10000" || response["responseCode"] == "10100"
+        (response["responseCode"] == "10000" && !(response["responseMessage"].include? "40")) || response["responseCode"] == "10100"
```

Comment:
How about `!response["responseMessage"].start_with?("40")` ? This puts the opening parenthesis for the second conditional method in a more clear place, and is more precise in matching the 40xxx codes at the start of the string.

Also it would be very helpful to codify the cases you mentioned in the pull request in the tests to prevent regressions. Could you add them to https://github.com/activemerchant/active_merchant/blob/master/test/unit/gateways/checkout_v2_test.rb ? Thanks!